### ISSUE TITLE
[a11y]: Screen readers miss status icon in InlineNotification

### ISSUE THREAD

**Issue by ritu-singh-vpc** [carbon-design-system/carbon#20348]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11.83.0

### Automated testing tool and ruleset

Success Criterion 1.4.3 Contrast (Minimum)

### Description

<img width="1915" height="1014" alt="Image" src="https://github.com/user-attachments/assets/e5f3b4d1-0119-45ab-afb7-8723d27bdfef" />

Accessibility Issue: Status Icon Not Announced to Screen Readers When hideCloseButton Is Used in InlineNotification.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-notifications-inline--default&args=hideCloseButton:!true;statusIconDescription:kkjjjjjjjj;title:Title

### Steps to reproduce

- Render an InlineNotification with kind="warning" and hideCloseButton.
- Provide minimal or no title/subtitle to simulate icon-only message.
- Use a screen reader (VoiceOver, NVDA, JAWS) to inspect the component.
- Observe that the "warning" intent is not conveyed at all.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
